File: LeetCode June Challenge/KthLargestElementinanArray.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> minHeap;
        if(nums.size() == 1) return nums[0];
        for(int i = 0; i < k; ++i) minHeap.push(nums[i]);
        for(int i = k; i < nums.size(); ++i){
            if(nums[i] > minHeap.top()){
                minHeap.pop();
                minHeap.push(nums[i]);
            }
        }
        return minHeap.top();
    }
};
int main(){
    Solution s;
    vector<int> nums = {1};
    int k = 1;
    cout<< s.findKthLargest(nums, k)<<endl;
    return 0;
}

File: Year-2023/LeetCode-August-Challenge/KthLargestElementinanArray.java
import java.util.*;

class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();

        for (int i = 0; i < k; ++i)
            minHeap.add(nums[i]);
        for (int i = k; i < nums.length; ++i) {
            if (nums[i] > minHeap.peek()) {
                minHeap.poll();
                minHeap.add(nums[i]);
            }
        }
        return minHeap.peek();
    }
}

public class KthLargestElementinanArray {
    public static void main(String[] args) {
        int[] nums = { 3, 2, 1, 5, 6, 4 };
        int k = 2;
        System.out.println(new Solution().findKthLargest(nums, k));
    }
}